c-dependencies/js-compute-runtime/builtins/object-store.cpp

```diff
@@ -175,6 +175,54 @@ bool parse_and_validate_key(JSContext *cx, JS::UniqueChars *key, size_t len) {
 
 } // namespace
 
+fastly_pending_object_store_lookup_handle_t ObjectStore::pending_lookup_handle(JSObject *self) {
+  MOZ_ASSERT(ObjectStore::is_instance(self));
+
+  JS::Value handle_value =
+      JS::GetReservedSlot(self, static_cast<uint32_t>(Slots::PendingLookupHandle));
+
+  if (handle_value.isInt32()) {
+    return handle_value.toInt32();
+  }
+
+  return INVALID_HANDLE;
+}
+
+bool ObjectStore::process_pending_object_store_lookup(JSContext *cx, JS::HandleObject self) {
+  MOZ_ASSERT(ObjectStore::is_instance(self));
+
+  fastly_option_body_handle_t ret;
+  fastly_error_t err;
+  bool ok = fastly_object_store_lookup_wait(builtins::ObjectStore::pending_lookup_handle(self),
+                                            &ret, &err);
+  if (!ok) {
+    HANDLE_ERROR(cx, err);
```

Comment:
Should this reject the promise?

Reply:
Yes this should reject rather than create an exception, good work spotting this bug!